### PR TITLE
Chart hints

### PR DESCRIPTION
**Pull request by waterplea**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes # 

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/core/components/tooltip/tooltip.component.ts [hunk 1]
```diff
@@ -65,8 +64,4 @@ export class TuiTooltipComponent extends TuiHintControllerDirective {
             event.stopPropagation();
         }
     }
-
-    get icon(): PolymorpheusContent {
-        return this.options.icon;
```
Comment (nsbarsukov) [inline projects/core/components/tooltip/tooltip.component.ts:70]: tooltip

Comment (nsbarsukov) [reply projects/core/components/tooltip/tooltip.component.ts:70]: https://taiga-ui--pr2273-chart-hints-5rshlhio.web.app/components/tooltip#options
vs
https://taiga-ui.dev/components/tooltip#options

File: projects/core/directives/hint-controller/hint-controller.provider.ts [hunk 2]
```diff
@@ -1,7 +1,15 @@
-import {ChangeDetectorRef, InjectionToken, Provider} from '@angular/core';
+import {
+    ChangeDetectorRef,
+    InjectionToken,
+    Optional,
+    Provider,
+    SkipSelf,
+} from '@angular/core';
 import {TuiDestroyService} from '@taiga-ui/cdk';
+import {TUI_HINT_OPTIONS, TuiHintOptions} from '@taiga-ui/core/directives';
```
Comment (nsbarsukov) [inline projects/core/directives/hint-controller/hint-controller.provider.ts:9]: it breaks build

File: projects/core/components/tooltip/tooltip.component.ts (hunk 1 above)
Comment (splincode) [reply projects/core/components/tooltip/tooltip.component.ts:70]: Also in dark theme

image

File: projects/demo/src/modules/charts/bar-chart/bar-chart.template.html [hunk 3]
```diff
@@ -31,6 +31,16 @@
             heading="Same values with collapsed mode"
             [content]="example2"
         >
+            <tui-notification class="tui-space_bottom-6">
+                Use
+                <a
+                    routerLink="/directives/hint-controller"
+                    tuiLink
+                >
+                    <code>HintController</code>
```
Comment (nsbarsukov) [inline projects/demo/src/modules/charts/bar-chart/bar-chart.template.html:40]: Let's add this controller to this component ?

https://github.com/Tinkoff/taiga-ui/blob/main/projects/demo/src/modules/app/autogenerated-code-hints/autogenerated-code-hints.component.ts

File: projects/demo/src/modules/components/abstract/inherited-documentation/inherited-documentation.template.html [hunk 4]
```diff
@@ -114,31 +114,25 @@
...
-            Hint mode
-        </ng-template>
-    </tui-doc-documentation>
-</ng-container>
+<tui-doc-documentation *ngIf="isTuiHint(documentedComponent)">
+    <ng-template
+        i18n
+        documentationPropertyName="tuiHintDirection"
+        documentationPropertyMode="input"
+        documentationPropertyType="TuiDirection"
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/abstract/inherited-documentation/inherited-documentation.template.html:122]:
```suggestion
        documentationPropertyType="TuiHintDirection"
```

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo/src/modules/components/abstract/inherited-documentation/inherited-documentation.template.html (hunk 4 above)
Comment (waterplea) [reply projects/demo/src/modules/components/abstract/inherited-documentation/inherited-documentation.template.html:122]: Stupid rebase. I wonder what else have I lost :(

File: projects/core/components/tooltip/tooltip.component.ts (hunk 1 above)
Comment (waterplea) [reply projects/core/components/tooltip/tooltip.component.ts:70]: Fixed :)

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**